src/ActionsImporter/Services/ProcessService.cs

```diff
@@ -18,8 +18,8 @@ public class ProcessService : IProcessService
         using var process = GetProcess(filename, arguments, cwd, environmentVariables);
         process.Start();
 
-        ReadStream(process.StandardOutput, output, cts.Token);
-        ReadStream(process.StandardError, output, cts.Token);
+        await ReadStream(process.StandardOutput, output, cts.Token);
+        await ReadStream(process.StandardError, output, cts.Token);
```

Comment:
I'm not entirely sure this is what we want. This would prevent StdErr from being read and relayed to the user's console until the StdOut stream is closed. Shouldn't these tasks be getting cleaned up when the cancellation token is cancelled below?

Reply:
Yeah I guess you're right, it's tricky indeed. But our old way is just force creating a thread with `Task.Run` and if anything happens during the reading of the console out/error no errors get returned. That being said the `ReadStream` is only supposed to read the stream so the chances of anything happening there is very minimal.
I'll look more into this and will try to address the issues.

Reply:
```csharp
        var stdOutTask = ReadStream(process.StandardOutput, output, cts.Token);
        var stdErrorTask = ReadStream(process.StandardError, output, cts.Token);

        await process.WaitForExitAsync(cts.Token);

        cts.Cancel();
        await Task.WhenAll(stdOutTask, stdErrorTask);

        if (process.ExitCode != 0)
        {
            var error = await process.StandardError.ReadToEndAsync();
            throw new Exception(error);
        }
```

Maybe we could get around this by starting the tasks but not awaiting them until the process exits? We'd need to confirm ☝️ starts streaming output right away though..

Reply:
This works but not the way we intend to. I think we can just use the recommended approach by using [OutputDataReceived](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.outputdatareceived?view=net-7.0) and `ErrorDataReceived` events, something like this:
```c#
var p = Process.Start(startInfo);

var outputData = new StringBuilder();
var errorData = new StringBuilder();

p.OutputDataReceived += (_, eventArgs) =>
{
    outputData.AppendLine(eventArgs.Data);
    Console.WriteLine($"Output: {eventArgs.Data}");
};

p.ErrorDataReceived += (_, eventArgs) =>
{
    errorData.AppendLine(eventArgs.Data);
    Console.WriteLine($"Output: {eventArgs.Data}");
};

p.BeginErrorReadLine();
p.BeginOutputReadLine();

await p.WaitForExitAsync();
```

this way we can also get rid of the `ReadStream` method altogether.

Reply:
We used to do something similar to [this](https://github.com/github/gh-valet/pull/11/files), but IIRC moved away from it because it didn't work well with unicode coloring and text replacement. Definitely worth validating though!